### ISSUE TITLE
Rarity Generator issues

### ISSUE THREAD

**Issue by kFuQ** [IAmJaysWay/Rarity-Ranking-NFT#4]
The rarity generator just stops running randomly and does not complete a cycle of 10K (BAYC).  I have to re-start it repeatedly.  Node just dies off and doesn't give you any clue about why.  I have tried adjusting the timer to 30sec and still get timeouts.

Another issue i have noticed is that if you restart the rarity generator, it will add duplicates to the moralis database.   I think it should have a checking feature where it would see if that tokenId exists in the database

Thanks for a cool tool and for the vids.  :D

**Comment by ericlam1114** [comment]
@kFuQ my computer did that too. Assuming you're talking about when node runs, the computer just stops running the operation after a couple thousand apes. I turned off the go-to-sleep mode on my computer and it fixed on my end. Basically just run the program and then leave my computer for 20 mins. When the computer goes to sleep it would stop executing the program. Not sure if this is the problem you're having too but hopefully it helps.

**Comment by kFuQ** [comment]
I think part of it is rate limiting on moralis as well.   (1500req/min for free accts iirc)

when i restarted the tool, that's when i noticed that it was adding duplicates into the moralis database.

**Comment by IAmJaysWay** [comment]
Yeah there may be limits with free accounts, but feel free to fork this project and add checking logic to identify if a tokenId already exists in the database. 👍

**Comment by mattsegura** [comment]
> Yeah there may be limits with free accounts, but feel free to fork this project and add checking logic to identify if a tokenId already exists in the database. 👍

any way to increase the runtime when processing the metadata?

**Comment by kFuQ** [comment]
i put checking into mine, it seems to work ok.  my biggest issue currently is with moralis authentication and creating and setting acl's so the whole table isn't public r/w.  

it also seems like node hangs up sometimes during nft[arr] loop and just stalls until you kill it off
